### ISSUE TITLE
Formalize OAB API by leveraging standard specification

### ISSUE THREAD

**Issue by bsavage** [OAButton/discussion#932]
## Feature request

An api specification format such as swagger 2.0 would be useful for standardizing doc
and for formalizing the api.  
There are opportunities to allow users to not only to interact with a nice gui into the documentation but
try out api operations interactively if the to-be-defined spec was freely hosted on an interactive site host (e.g., swaggerhub).  Would serve a variety of use cases, including those below.

## Use case template
* **Open tool developer** wants to **consume a standard api spec** because **s/he needs to quickly get up to speed with its usage**
* **Open tool developer** wants to **try out api** because **s/he needs to prove it can be integrated**
* **Open tool developer** wants to **discover apis** because **they want to leverage other works.**
* **OAB team** wants to **provide a standardized api format** because **it should make it easier to maintain.**

**Comment by JosephMcArthur** [comment]
Thanks for posting this @bsavage! 

@markmacgillivray would love your thoughts here, although I know you've got a bunch on at the moment and shipping the new build of find & ILL tools is priority. 

Just to be clear (I'm not super familiar with Swagger). This is about better documentation, rather than changing the API itself. Is that right?

**Comment by bsavage** [comment]
@JosephMcArthur 
Yes, to present the api in a more structured, standard way - for potential use case benefits above.  The API is not required to change.  Some consideration might be given to hosting the doc at a swagger-aware host (either a freemium one, or by standing up one) - a less optimal alternative (in terms of consumers) is to generate static html pages or simply provide the spec as a link someone can load into an intelligent api tool.

**Comment by markmacgillivray** [comment]
I'm not against someone changing the API docs if desired, but it is not something I would want to contribute time to, as I don't feel the benefits are worth the effort.

Particularly in the case of the OAB API:

* there are only 8 public endpoints and it is very easy to get up to speed with
* it can already be easily tried out regardless of what format the docs are in - tools to try APIs have to be used when it comes to using APIs anyway
* it can be discovered regardless of format - if you want more discovery, there are many ways to advertise it regardless of docs format
* I think it is easier to maintain the docs we have than to maintain Swagger docs

Using something like Swagger would reduce the amount of people who could potentially write the docs to those that know the standard, and require anyone wishing to contribute to maintaining the docs to learn that standard first - all of which sounds like more barriers to maintenance than just writing them. Using an interactive API doc tool would also require ensuring that our docs continue to get updated and deployed to such a tool, or that we run one ourselves. None of these things are hard, but I don't believe they return sufficient benefits compared to the simple page we have now, so as mentioned above I'm not against it if someone wants to take on the work of doing it, but I don't want to do that.
